graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/migration/CassandraSchemaHelper.java

```diff
@@ -0,0 +1,222 @@
+            .bind(),
+        "Drop table " + table.name(),
+        false);
+  }
+
+  public static MigrationQuery buildAddColumnQuery(
+      Table table, Column column, DataStore dataStore) {
+    return new MigrationQuery(
+        dataStore
+            .queryBuilder()
+            .alter()
+            .table(table.keyspace(), table.name())
+            .addColumn(column)
+            .build()
+            .bind(),
+        String.format("Add table column %s.%s", table.name(), column.name()),
+        true);
+  }
+
+  /** @return a list of differences, or empty if the UDTs match. */
+  public static List<Difference> compare(UserDefinedType expectedType, UserDefinedType actualType) {
+
+    // Note: we deliberately avoid comparing the keyspaces, they might differ because of keyspace
+    // decoration.
+
+    String typeName = expectedType.name();
+
+    if (!typeName.equals(actualType.name())) {
+      throw new IllegalArgumentException("This should only be called for UDTs with the same name");
+    }
+
+    List<Difference> differences = new ArrayList<>();
+    for (Column expectedColumn : expectedType.columns()) {
+      Column actualColumn = actualType.columnMap().get(expectedColumn.name());
+      compareColumn(expectedColumn, actualColumn, differences);
+    }
+    return differences;
+  }
+
+  public static MigrationQuery buildCreateQuery(UserDefinedType type, DataStore dataStore) {
+    return new MigrationQuery(
+        dataStore.queryBuilder().create().type(type.keyspace(), type).build().bind(),
+        "Create UDT " + type.name(),
+        false);
+  }
+
+  public static MigrationQuery buildDropQuery(UserDefinedType type, DataStore dataStore) {
+    return new MigrationQuery(
+        dataStore.queryBuilder().drop().type(type.keyspace(), type).ifExists().build().bind(),
+        "Drop UDT " + type.name(),
+        false);
+  }
+
+  public static MigrationQuery buildAddColumnQuery(
+      UserDefinedType type, Column column, DataStore dataStore) {
+    return new MigrationQuery(
+        dataStore
+            .queryBuilder()
+            .alter()
+            .type(type.keyspace(), type)
+            .addColumn(column)
+            .build()
+            .bind(),
+        String.format("Add UDT field %s.%s", type.name(), column.name()),
+        true);
+  }
+
+  public enum DifferenceType {
+    MISSING_COLUMN,
+    WRONG_TYPE,
+    WRONG_KIND,
+    WRONG_CLUSTERING_ORDER,
+  }
+
+  public static class Difference {
+
+    private final Column column;
+    private final DifferenceType type;
+    private final String description;
+
+    public Difference(Column column, DifferenceType type, String description) {
+      this.column = column;
+      this.type = type;
+      this.description = description;
+    }
+
+    public Column getColumn() {
+      return column;
+    }
+
+    public DifferenceType getType() {
+      return type;
+    }
+
+    @Override
+    public String toString() {
+
+      return String.format(
```

Comment:
This `toString` seems important because we are using it in the logic. Should we instead create a custom method that returns the representation of this class? (I don't strongly feel about it)

Reply:
Good point, will do.